src/app/nhs-app-template-submitted/[sessionId]/page.tsx

```diff
@@ -1,9 +1,32 @@
 'use server';
 
-const NhsAppTemplateSubmittedPage = async () => (
-  <h2 className='nhsuk-heading-l' data-testid='page-sub-heading'>
-    Placeholder page
-  </h2>
-);
+import { randomUUID } from 'node:crypto';
+import { redirect, RedirectType } from 'next/navigation';
+import { getSession, sendEmail } from '@utils/form-actions';
+import { PageProps } from '@utils/types';
+
+const NhsAppTemplateSubmittedPage = async ({
+  params: { sessionId },
+}: PageProps) => {
+  const session = await getSession(sessionId);
+
+  if (!session) {
+    redirect('/invalid-session', RedirectType.replace);
+  }
+
+  const templateId = randomUUID();
+
+  await sendEmail(
```

Comment:
Wouldn't this resend the email on every page refresh? 

I was finding a similar thing on the submitted flow. I've put the DB save on the form post from the previous page.

Reply:
as discussed, your approach of having the actions be on the form submit for the previous page does work much better, but i don't have the work you've done in that area on this branch.

if your work goes in before this, i'll move the sendEmail call to go with where you've put the save template stuff, but if this goes in first you'll have to move it

Reply:
This will probably be in before my PR. I can move it over :)